### PR TITLE
aegisub: 3.3.3 -> feature_11

### PR DESCRIPTION
**Pull request by Hobr**
## Description of changes

The old package is based on [wangqr/Aegisub](https://github.com/wangqr/Aegisub), which already haven't update for 2 years.Now most of people use [arch1t3cht/Aegisub](https://github.com/arch1t3cht/Aegisub) version, it updates frequently with more features.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by wegank** [comment]
It's not very nice to see me and @AndersonTorres silently removed from the maintainers....

File: pkgs/applications/video/aegisub/default.nix
Comment (wegank) [inline pkgs/applications/video/aegisub/default.nix:164]: Why not `lib.mesonEnable "alsa" alsaSupport`?

File: pkgs/applications/video/aegisub/default.nix
Comment (wegank) [inline pkgs/applications/video/aegisub/default.nix:130]: Does `NIX_LDFLAGS` work? `NIX_CFLAGS_LINK` is somewhat undocumented.

File: pkgs/applications/video/aegisub/default.nix
Comment (wegank) [inline pkgs/applications/video/aegisub/default.nix:205]: Can it be put in a patch using `substitute`?

File: pkgs/applications/video/aegisub/default.nix
Comment (wegank) [inline pkgs/applications/video/aegisub/default.nix:106]: Please place the reformatting in a separate commit before the changes.

(Nor does it follow [RFC 166](https://github.com/NixOS/rfcs/blob/master/rfcs/0166-nix-formatting.md): do you use formatters other than `nixfmt-rfc-style`?)

File: pkgs/applications/video/aegisub/default.nix
Comment (wegank) [inline pkgs/applications/video/aegisub/default.nix:124]: I don't see the darwin frameworks any more: did you drop darwin support?

File: pkgs/applications/video/aegisub/0001-bas-to-bs.patch
Comment (wegank) [inline pkgs/applications/video/aegisub/0001-bas-to-bs.patch:9]: Please use `fetchpatch` instead of putting this in the tree.

File: pkgs/applications/video/aegisub/default.nix
Comment (wegank) [inline pkgs/applications/video/aegisub/default.nix:53]: Please provide links to find these revisions, for example

```suggestion
    # https://github.com/arch1t3cht/Aegisub/blob/feature_11/subprojects/bestsource.wrap#L3
    rev = "ba1249c1f5443be6d0ec2be32490af5bbc96bf99";
```

**Review by wegank — CHANGES_REQUESTED** [review]
I'll continue the review once the comments are resolved; for now, I'm blocking the PR because ofborg isn't happy with the darwin builds.

**Comment by Hobr** [comment]
> It's not very nice to see me and @AndersonTorres silently removed from the maintainers....

I'm sorry for I have some misunderstand for the 'maintainers'.... I just thought it referred to the maintainer of the upstream repo's contributors.... I will add this part back to code and try to meet the merge requirement then request for merge again:)

**Comment by AndersonTorres** [comment]
I am in faraway mode. Since wegank captured the truckload of problems, I will wait.

**Comment by Hobr** [comment]
Wrongly merge upstream:)I will reopen PR.

**Comment by drupol** [comment]
Please open it in **draft** by default to avoid this kind of issues.

**Pull request closed without merging**
